hardhat.config.ts

```diff
@@ -84,4 +78,19 @@ const config: HardhatUserConfig = {
   },
 };
 
+if (process.env.TEST) {
+  const solidity = config.solidity as MultiSolcUserConfig;
+  solidity.compilers.forEach((_, i) => {
+    solidity.compilers[i].settings! = {
+      ...solidity.compilers[i].settings!,
+      outputSelection: {
+        '*': {
+          '*': ['storageLayout'],
+        },
+      },
+    };
+  });
+  config.solidity = solidity;
```

Comment:
I believe it would be nicer to use a map, maybe this works (I didn't compile it, so not sure)
```
config.solidity.compilers = config.solidity.compilers.map(compiler => ({
    ...compiler,
    settings: {
        ...compiler.settings,
        outputSelection: {
            '*': {
                '*': ['storageLayout'],
            },
        },
    }
}));
```